src/actions/usersActions.js

```diff
@@ -0,0 +1,36 @@
+import server from '../Api/server';
+import * as types from '../actionTypes/usersTypes';
+
+/**
+ * Toggles the loading state
+ * @return {object} action
+ */
+export const toggleLoading = (state) => ({
+  type: types.TOGGLE_LOADING,
+  payload: state,
+});
+
+/**
+ * Set users
+ * @return {object} action
+ */
+export const setUsers = (users) => ({
+  type: types.SET_USERS,
+  payload: users,
+});
+
+/**
+ * `Fetch users`
+ * @return {object} action
+ */
+export const getLatestUsers = () => async (dispatch) => {
+  dispatch(toggleLoading(true));
+
+  try {
+    const { data: { profiles: users } } = await server.get('/profiles');
+    dispatch(toggleLoading(false));
+    dispatch(setUsers(users));
+  } catch (e) {
+    dispatch(toggleLoading(false));
+  }
```

Comment:
Do you think that `try and catch` is really necessary here? https://blog.grossman.io/how-to-write-async-await-without-try-catch-blocks-in-javascript

Reply:
@oesukam In case of an error such as an internet connection error, or in case the server is down, this will prevent the application from crashing, and we can return a custom error message.

Reply:
@gracelungu Not really, `try and catch` will still wait for the request rejection. You can have a look at the link I shared